File: 2024/01/solution.exs
defmodule Solution do
  def part_one(left, right) do
    Enum.zip(Enum.sort(left), Enum.sort(right))
      |> Enum.map(fn {a, b} -> abs(a - b) end)
      |> Enum.sum
      |> IO.inspect
  end

  def part_two(left, right) do
    Enum.map(left, fn a -> Enum.count(right, fn b -> a == b end) * a end)
      |> Enum.sum
      |> IO.inspect
  end

  def run do
    [left, right] = __DIR__ <> "/" <> "input.txt"
      |> File.read!()
      |> String.trim()
      |> String.split("\n")
      |> Enum.map(fn s -> Enum.map(String.split(s, "   "), &String.to_integer/1) end)
      |> Enum.zip_with(&Function.identity/1)
    part_one(left, right)
    part_two(left, right)
  end
end

IO.write("\e[H\e[2J")
IO.puts("Running #{__ENV__.file}...")
Solution.run

File: 2024/01/solution.py
input = open("2024/01/input.txt", "r")

input = map(lambda line: map(int, line.strip().split('   ')), input.readlines())

a = []
b = []

for pair in input:
    a.append(pair[0])
    b.append(pair[1])

a.sort()
b.sort()

part_one = 0

for pair in zip(a, b):
    part_one += abs(pair[1] - pair[0])

print part_one

part_two = 0

for m in a:
    part_two += b.count(m) * m

print part_two
